examples/chat_with_document/handler.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2023 The KubeAGI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/gofiber/fiber/v2"
+	zhipuaiembeddings "github.com/kubeagi/arcadia/pkg/embeddings/zhipuai"
+	"github.com/tmc/langchaingo/documentloaders"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/textsplitter"
+
+	"github.com/kubeagi/arcadia/pkg/llms/zhipuai"
+	"github.com/kubeagi/arcadia/pkg/vectorstores/chromadb"
+)
+
+type Workload struct {
+	Document     string `json:"document"`
+	Namespace    string `json:"namespace"`
+	ChunkSize    int    `json:"chunk-size"`
+	ChunkOverlap int    `json:"chunk-overlap"`
+}
+
+func HomePageGetHandler(c *fiber.Ctx) error {
+	return c.SendString("This is the home page of chat server sample. Send POST request to /chat with your question to chat with me!")
+}
+
+func LoadHandler(c *fiber.Ctx) error {
+	// Convert body to json workload
+	var workload Workload
+	err := c.BodyParser(&workload)
+	if err != nil {
+		return errors.New("error parsing body to workload type" + err.Error())
+	}
+
+	if workload.Document == "" {
+		return errors.New("document cannot be empty")
+	}
+
+	var docReader io.Reader
+
+	_, err = docReader.Read([]byte(workload.Document))
+
+	if err != nil {
+		return errors.New("Error reading document:" + err.Error())
+	}
+
+	loader := documentloaders.NewText(docReader)
+
+	split := textsplitter.NewRecursiveCharacter()
+	split.ChunkSize = workload.ChunkSize
+	split.ChunkOverlap = workload.ChunkOverlap
+
+	documents, err := loader.LoadAndSplit(context.Background(), split)
+	if err != nil {
+		return errors.New("Error loading documents:" + err.Error())
+	}
+
+	err = workload.EmbedDocument(context.Background(), documents)
+	if err != nil {
+		return errors.New("Error embedding documents:" + err.Error())
+	}
+
+	return c.SendString("OK")
+}
+
+func (w Workload) EmbedDocument(ctx context.Context, documents []schema.Document) error {
```

Comment:
If you want to use `(w Workload)`, you can move `document load` and `splt` in this function